### PR TITLE
Fix for Reduce operators

### PR DESCRIPTION
**Pull request by anirudhacharya**
Models with reduce operators were not getting imported into mxnet, as described in this issue. This PR fixes that issue and has unit tests to cover those scenarios.

### CODE REVIEW

**Comment by anirudhacharya** [comment]
@Roshrini

**Comment by lupesko** [comment]
Looks like there is an accuracy issue with reduce_sum

File: onnx_mxnet/backend.py [hunk 1]
```diff
@@ -79,14 +81,14 @@ def run_node(cls, node, inputs, device='CPU'):
...
-            if node.op_type == 'Slice' or node.op_type == 'Pad':
+            if node.op_type in reduce_op_types:
                 data_forward.append(mx.nd.array(val))
             else:
                 data_forward.append(mx.nd.array([val]))
 
         mod.forward(batch(data_forward))
         result = mod.get_outputs()[0].asnumpy()
-        if node.op_type == 'Slice' or node.op_type == 'Pad':
+        if node.op_type in reduce_op_types:
```
Comment (spidyDev) [inline onnx_mxnet/backend.py]: Do we need to add extra dimension to the "result", and in the unit  test we use [0] to compare. I think not adding extra [] ( [result] ), and then just comparing the whole output with the numpy result should do the trick.
I do not know the reason behind doing this for "slice"/"pad" , probably @Roshrini  can clarify.

Comment (Roshrini) [reply onnx_mxnet/backend.py]: Added extra dimension in the result for "slice"/"pad" to match the output with onnx_test_backend as MXNetBackend class in backend.py is used for onnx test infrastructure as well. We are planning to rely on onnx_test_backend for all the operators test(removing test_layers.py). So when we test the backend for all operators, this logic can be changed.

**Pull request merged**